File: src/main/java/com/back/mozu/global/config/SchedulerConfig.java
package com.back.mozu.global.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.TaskScheduler;
import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;

@Configuration
public class SchedulerConfig {

    @Bean
    public TaskScheduler taskScheduler() {
        ThreadPoolTaskScheduler scheduler = new ThreadPoolTaskScheduler();
        scheduler.setPoolSize(10); // 동시에 처리할 수 있는 Task 수
        scheduler.setThreadNamePrefix("release-scheduler-");
        scheduler.initialize();
        return scheduler;
    }
}

File: src/main/java/com/back/mozu/global/config/SchedulerConfig.kt
package com.back.mozu.global.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.TaskScheduler
import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler

@Configuration
class SchedulerConfig {
    @Bean
    fun taskScheduler(): TaskScheduler {
        val scheduler = ThreadPoolTaskScheduler()
        scheduler.setPoolSize(10)
        scheduler.setThreadNamePrefix("release-scheduler-")
        scheduler.initialize()
        return scheduler
    }
}
